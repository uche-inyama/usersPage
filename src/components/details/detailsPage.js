import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import { sheduleMeeting } from '../../action';
import DetailWrapper from './detailStyle';

export const DetailPage = ({ centers, bookings }) => {
  const { id } = useParams();
  const [email, setEmailAddress] = useState('');
  const [useCity, setCity] = useState('');

  const center = centers.find(element => element.id === parseInt(id, 10));
  const image = center ? center.image : '';
  const hall = center ? center.hall : '';
  const building = center ? center.building : '';
  const price = center ? center.price : '';
  const capacity = center ? center.capacity : '';
  const state = center ? center.state : '';
  const city = center ? center.city : '';

  const url = `${image}`;

  const currentUser = localStorage.getItem('current_user');

  const emailChangeHandler = e => {
    e.preventDefault();
    setEmailAddress(e.target.value);
  };

  const cityChangeHandler = e => {
    e.preventDefault();
    setCity(e.target.value);
  };

  const submitHandler = e => {
    e.preventDefault();
    const data = new FormData(e.target);
    bookings(data);
    setEmailAddress('');
    setCity('');
  };

  return (
    <DetailWrapper>
      <div className="top-wrapper">
        <div className="image-wrapper">
          <div
            className="image"
            style={{
              backgroundImage: `url(${url})`,
            }}
          />
        </div>
        <div className="details">
          <ul>
            <li>
              Building:
              {building}
            </li>
            <li>
              Location:
              {city}
              {' '}
              city
              {' '}
            </li>
            <li>
              State:
              {state}
            </li>
            <li>
              Capacity:
              {capacity}
            </li>
            <li>
              Price:
              {price}
            </li>
          </ul>
        </div>
      </div>

      <div className="book-appointment">
        <h2>Book an Appointment with us.</h2>
        <form onSubmit={submitHandler}>
          <input
            type="text"
            name="appointment[username]"
            value={currentUser}
            readOnly
          />
          <br />

          <input
            type="text"
            name="appointment[hall]"
            value={center ? center.hall : ''}
            readOnly
          />
          {' '}
          <br />

          <input
            type="text"
            name="appointment[email]"
            onChange={emailChangeHandler}
            placeholder="email"
            value={email}
          />
          <br />
          <input
            type="text"
            name="appointment[city]"
            onChange={cityChangeHandler}
            placeholder="city"
            value={useCity}
          />
          <br />
          <input type="submit" value="submit" />
        </form>
      </div>
    </DetailWrapper>
  );
};

DetailPage.propTypes = {
  centers: PropTypes.arrayOf(
    PropTypes.shape({
      building: PropTypes.string.isRequired,
      hall: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      state: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      capacity: PropTypes.number.isRequired,
    }),
  ).isRequired,
  bookings: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  centers: state.centers,
});

const mapDispatchToProps = dispatch => ({
  bookings: (data => {
    dispatch(sheduleMeeting(data));
  }),
});

export default connect(mapStateToProps, mapDispatchToProps)(DetailPage);
