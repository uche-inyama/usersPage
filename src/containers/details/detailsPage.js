import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import { sheduleMeeting, removeUser } from '../../actions/action';
import DetailWrapper from './detailStyle';
import NavBar from '../../components/navBar/navBar';

const DetailPage = ({
  centers, bookings, match, logOut, currentUser,
}) => {
  const { id } = match.params;
  const centerId = parseInt(id, 10);
  const [email, setEmailAddress] = useState('');
  const [useCity, setCity] = useState('');
  const [selectedDate, setSelectedDate] = useState(null);

  const center = centers.find(element => element.id === parseInt(id, 10));
  const image = center ? center.image : '';
  const hall = center ? center.hall : '';
  const building = center ? center.building : '';
  const price = center ? center.price : '';
  const capacity = center ? center.capacity : '';
  const state = center ? center.state : '';
  const city = center ? center.city : '';

  const url = `${image}`;

  const user = localStorage.getItem('current_user');
  const getUserId = localStorage.getItem('user_id');
  const userId = parseInt(getUserId, 10);

  const emailChangeHandler = e => {
    e.preventDefault();
    setEmailAddress(e.target.value);
  };

  const cityChangeHandler = e => {
    e.preventDefault();
    setCity(e.target.value);
  };

  const submitHandler = e => {
    e.preventDefault();
    const form = document.querySelector('form');
    const data = new FormData(e.target);
    bookings(data);
    setEmailAddress('');
    setCity('');
    setSelectedDate('');
    form.style.display = 'none';
  };

  const handleBookAppointments = () => {
    const form = document.querySelector('form');
    form.style.display = 'block';
  };

  const renderNavBar = user ? (
    <NavBar
      currentUser={currentUser}
      logOut={logOut}
    />
  ) : ('');

  return (
    <DetailWrapper>
      {renderNavBar}
      <main className="DetailsWrapper">
        <section className="top-wrapper">
          <div className="image-wrapper">
            <div
              className="image"
              style={{
                backgroundImage: `url(${url})`,
              }}
            />
          </div>
          <aside className="details">
            <div className="head">
              <h1>{hall}</h1>
              <h6>Lorem ipsum dolor sit amet</h6>
            </div>
            <ul>
              <li className="item-detail">
                <span>Building </span>
                {' '}
                <span>{building}</span>
              </li>
              <li className="item-detail">
                <span> City </span>
                {' '}
                <span>{city}</span>
              </li>
              <li className="item-detail">
                <span>State </span>
                {' '}
                <span>{state}</span>
              </li>
              <li className="item-detail">
                <span>Capacity </span>
                {' '}
                <span>{capacity}</span>
              </li>
              <li className="item-detail">
                <span>Price </span>
                {' '}
                <span>
                  {price}
                  {' '}
                  units
                </span>
              </li>
            </ul>
          </aside>
        </section>

        <section className="book-appointment">
          {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
          <h2 onClick={handleBookAppointments}>Book an Appointment with us.</h2>
          <form onSubmit={submitHandler}>
            <input
              type="hidden"
              name="appointment[center_id]"
              value={centerId}
            />
            <input
              type="hidden"
              name="appointment[user_id]"
              value={userId}
            />
            <input
              type="text"
              name="appointment[username]"
              value={user}
              readOnly
            />
            <br />

            <input
              type="text"
              name="appointment[hall]"
              value={hall}
              readOnly
            />
            {' '}
            <br />

            <input
              type="text"
              name="appointment[email]"
              onChange={emailChangeHandler}
              placeholder="email"
              value={email}
            />
            <br />
            <input
              type="text"
              name="appointment[city]"
              onChange={cityChangeHandler}
              placeholder="city"
              value={useCity}
            />
            <br />
            <DatePicker
              selected={selectedDate}
              onChange={date => setSelectedDate(date)}
              placeholderText="date"
              name="appointment[date]"
              value={selectedDate}
            />
            <br />
            <input type="submit" value="submit" />
          </form>
        </section>
      </main>
    </DetailWrapper>
  );
};

DetailPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  centers: PropTypes.arrayOf(
    PropTypes.shape({
      building: PropTypes.string.isRequired,
      hall: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      state: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      capacity: PropTypes.number.isRequired,
    }),
  ).isRequired,
  bookings: PropTypes.func.isRequired,
  logOut: PropTypes.func.isRequired,
  currentUser: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  centers: state.centers,
  currentUser: state.currentUser,
});

const mapDispatchToProps = dispatch => ({
  bookings: (data => {
    dispatch(sheduleMeeting(data));
  }),
  logOut: (() => {
    dispatch(removeUser());
  }),
});

export default connect(mapStateToProps, mapDispatchToProps)(DetailPage);
