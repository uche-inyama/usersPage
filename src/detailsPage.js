import React, { useState } from 'react';
import { connect } from 'react-redux'
import { useParams } from 'react-router-dom';
import { sheduleMeeting } from './action'

const DetailPage = ({ centers, bookings }) => {
  let { id } = useParams();

  const [email, setEmailAddress] = useState('')
  const [city, setCity] = useState('')

  const center = centers.find(element => element.id === parseInt(id))
  const username = localStorage.getItem('username');
  const item = localStorage.getItem('appointment')

  console.log(item)
  const emailChangeHandler = (e) => {
    e.preventDefault()
    setEmailAddress(e.target.value)
  }

  const cityChangeHandler = (e) => {
    e.preventDefault()
    setCity(e.target.value)
  }

  const submitHandler = (e) => {
    e.preventDefault()
    const data = new FormData(e.target);
    bookings(data);
    setEmailAddress('');
    setCity('');
  }

  return (
    <div>
      <div className="image-wrapper">

      </div>
      <div className="book-appointment">
        <form onSubmit={submitHandler}>
          <input type="text"
            name="appointment[username]"
            value={username}
            readOnly={true}
          />

          <input type="text"
            name="appointment[hall]"
            value={center ? center.hall : ''}
            readOnly={true}
          />

          <input type="text"
            name="appointment[email]"
            onChange={emailChangeHandler}
            placeholder='email'
            value={email}
          />
          <input type="text"
            name="appointment[city]"
            onChange={cityChangeHandler}
            placeholder='city'
            value={city}
          /><br />
          <input type="submit" value="submit" />
        </form>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    centers: state.centers
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    bookings: ((data) => {
      dispatch(sheduleMeeting(data))
    })
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(DetailPage);