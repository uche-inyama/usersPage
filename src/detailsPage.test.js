import React from 'react'
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import DetailsPage from './detailsPage';
import { DetailPage } from './detailsPage'


const mockStore = configureStore();
const store = mockStore();

const props = {
  centers: [{ building: "Alvan", hall: "Albion" }]
};

describe('detailsPage as a connected component', () => {
  it('', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <DetailsPage {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  })

  it.only('', () => {
    const dispatch = jest.fn();
    store.dispatch = dispatch();

    const wrapper = shallow(
      <Provider store={store}>
        <DetailsPage />
      </Provider>
    );
    expect(dispatch).toHaveBeenCalledTimes(1);
  })
  it('', () => {
    const onChange = jest.fn();

    const wrapper = shallow(
      <DetailPage {...props} onChange={onChange} />
    );

    let nameInput = wrapper.find('input').first()
    nameInput.simulate('change', {
      target: { value: 'Jack' },
    })
    nameInput = wrapper.find('input').first()
    expect(nameInput.props().value).toEqual('Jack')
  })
})