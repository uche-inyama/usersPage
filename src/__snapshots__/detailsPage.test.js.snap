// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsPage as a connected component  1`] = `ShallowWrapper {}`;
